Mathlib/Probability/Moments.lean

```diff
@@ -333,4 +344,363 @@ theorem measure_le_le_exp_cgf [IsFiniteMeasure μ] (ε : ℝ) (ht : t ≤ 0)
+      have q00 : ‖rexp (t * X ω)‖ ≤ rexp (|t| * c) := by
+        simp only [norm_eq_abs, abs_exp, exp_le_exp]
+        calc
+        _ ≤ |t * X ω| := le_abs_self (t * X ω)
+        _ = |t| * |X ω| := abs_mul t (X ω)
+        _ ≤ |t| * c := by
+          apply mul_le_mul_of_nonneg_left
+          rw [← (by dsimp only [c]; simp only
+            [norm_eq_abs, abs_eq_self, le_max_iff, abs_nonneg, or_self] :
+            |c| = c)]
+          exact le_trans' (le_abs_self (max ‖a‖ ‖b‖)) (abs_le_max_abs_abs ha hb)
+          exact abs_nonneg t
+      exact q00
+  · dsimp only [e']
+    rw [aestronglyMeasurable_iff_aemeasurable]
+    apply AEMeasurable.mul
+    · apply Measurable.comp_aemeasurable' _ _
+      · exact measurable_exp
+      · exact AEMeasurable.const_mul hX t
+    exact AEMeasurable.pow_const hX 2
+  · set bound := fun ω ↦ rexp ((2 * |t| + 1) * |X ω|) * (|X ω| ^ 2)
+    suffices ∀ ω, ∀ x ∈ Metric.ball t (|t| + 1), ‖e' x ω‖ ≤ bound ω from ae_of_all μ this
+    intros ω x h
+    dsimp [e', bound]
+    simp only [sq_abs]
+    have h' : |rexp (x * X ω) * X ω ^ 2| ≤ rexp ((2 * |t| + 1) * |X ω|) * X ω ^ 2 := by
+      calc
+      _ = |rexp (x * X ω)| * |(X ω ^ 2)| := abs_mul (rexp (x * X ω)) (X ω ^ 2)
+      _ = rexp (x * X ω) * (X ω ^ 2) := by simp only [abs_exp, abs_pow, sq_abs]
+      _ ≤ rexp ((2 * |t| + 1) * |X ω|) * X ω ^ 2 := by
+        suffices rexp (x * X ω) ≤
+          rexp ((2 * |t| + 1) * |X ω|) from mul_le_mul_of_nonneg_right this (sq_nonneg (X ω))
+        simp only [exp_le_exp]
+        calc
+        _ ≤ |x * X ω| := le_abs_self (x * X ω)
+        _ = |x| * |X ω| := abs_mul x (X ω)
+        _ ≤ (2 * |t| + 1) * |X ω| := by
+          suffices |x| ≤ 2 * |t| + 1 from mul_le_mul_of_nonneg_right this (abs_nonneg (X ω))
+          simp only [Metric.mem_ball] at h
+          linarith [le_trans' (le_of_lt h) (abs_sub_abs_le_abs_sub x t)]
+    exact h'
+  · apply MeasureTheory.Integrable.bdd_mul'
+    · simp only [sq_abs]
+      rw [(by ext ω; ring : (fun ω ↦ X ω ^ 2) = (fun ω ↦ X ω * X ω))]
+      apply MeasureTheory.Integrable.bdd_mul'
+        (integrable_bounded a b X hX h) (aestronglyMeasurable_iff_aemeasurable.mpr hX)
+      · filter_upwards [ha, hb] with x ha hb
+        exact (by simp only [norm_eq_abs]; exact
+          le_trans' (le_abs_self (max ‖a‖ ‖b‖)) (abs_le_max_abs_abs ha hb) : ‖X x‖ ≤ |c|)
+    · apply aemeasurable_expt (fun ω ↦ |X ω|) (2 * |t| + 1) _
+      apply Measurable.comp_aemeasurable' _ hX
+      apply measurable_norm
+    · filter_upwards [ha, hb] with ω ha hb
+      have p0 : ‖rexp ((2 * |t| + 1) * |X ω|)‖ ≤ ‖rexp ((2 * |t| + 1) * c)‖ := by
+        simp only [norm_eq_abs, abs_exp, exp_le_exp]
+        rw [← (abs_eq_self.mpr (le_trans' (le_max_left |a| ‖b‖) (abs_nonneg a)) : |c| = c)]
+        exact mul_le_mul_of_nonneg_left
+          (le_trans' (le_abs_self (max ‖a‖ ‖b‖)) (abs_le_max_abs_abs ha hb))
+          (by linarith [abs_nonneg t] : 0 ≤ 2 * |t| + 1)
+      exact p0
```

Comment:
If you have a `have p0` followed by `exact p0`, then you could just prove it directly.

```suggestion
      change ‖rexp ((2 * |t| + 1) * |X ω|)‖ ≤ ‖rexp ((2 * |t| + 1) * c)‖  -- perhaps not needed
      simp only [norm_eq_abs, abs_exp, exp_le_exp]
      rw [← (abs_eq_self.mpr (le_trans' (le_max_left |a| ‖b‖) (abs_nonneg a)) : |c| = c)]
      exact mul_le_mul_of_nonneg_left
        (le_trans' (le_abs_self (max ‖a‖ ‖b‖)) (abs_le_max_abs_abs ha hb))
        (by linarith [abs_nonneg t] : 0 ≤ 2 * |t| + 1)
```